src/main/java/seedu/address/logic/commands/ViewCommand.java

```diff
@@ -24,89 +28,158 @@ public class ViewCommand extends Command {
+
+            return new EntryInYearMonthPredicate(yearMonth.get());
+        }
+
+        // By default, show all entries in the entry list.
+        return Model.PREDICATE_SHOW_ALL_ENTRIES;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        // model.setMonthForChart(month);
+
+        EntryType entryType = viewEntriesDescriptor.getEntryType();
+        GraphType graphType = viewEntriesDescriptor.getGraphType();
+        GraphConfiguration graphConfiguration = new GraphConfiguration(entryType, graphType, true);
+
+
+        Predicate<Entry> predicate = predicateSelector(viewEntriesDescriptor);
 
         switch (entryType.getEntryType()) {
         case EXPENDITURE:
-            switch (graphType.getGraphType()) {
-            case CATEGORY:
-                return new CommandResult(String.format(MESSAGE_SUCCESS, ENTRY_EXPENDITURE, GRAPH_CATEGORY), false,
-                                         false, true);
-            case MONTH:
-                return new CommandResult(String.format(MESSAGE_SUCCESS, ENTRY_EXPENDITURE, GRAPH_MONTH), false,
-                        false, true);
-            default:
-                //should never reach here
-                return null;
-            }
-
+            model.updateFilteredExpenditureList(predicate);
+            break;
         case INCOME:
-            switch (graphType.getGraphType()) {
-            case CATEGORY:
-                return new CommandResult(String.format(MESSAGE_SUCCESS, ENTRY_INCOME, GRAPH_CATEGORY), false, false,
-                                         true);
-            case MONTH:
-                return new CommandResult(String.format(MESSAGE_SUCCESS, ENTRY_INCOME, GRAPH_MONTH), false, false,
-                        true);
-            default:
-                //should never reach here
-                return null;
-            }
+            model.updateFilteredIncomeList(predicate);
+            break;
         default:
-            //should never reach here
-            return null;
+            break;
         }
+
+        return new CommandResult(
+                String.format(MESSAGE_SUCCESS, entryType, graphType), false, false, graphConfiguration);
     }
 
     @Override
     public boolean equals(Object other) {
         if (other == this) {
             return true;
         }
+        // instanceof handles nulls
         if (!(other instanceof ViewCommand)) {
             return false;
         }
         ViewCommand otherViewCommand = (ViewCommand) other;
-        if (this.month == null) {
-            return this.entryType.equals(otherViewCommand.entryType);
+        return viewEntriesDescriptor.equals(otherViewCommand.viewEntriesDescriptor);
+    }
+
+    /**
+     * Stores the details to view entries. Each non-empty field value will replace the
+     * default value of the view configuration.
+     */
+    public static class ViewEntriesDescriptor {
```

Comment:
Thanks for this! Should be a lot easier to explain in our DG as well. Think for Diagrams we can replicate what we did for the Edit command descriptor for the View Entryies